src/main/java/com/mbcoder/iot/gpslogger/GPS_Logger.java

```diff
@@ -0,0 +1,329 @@
+ * under the License.
+ */
+
+package com.mbcoder.iot.gpslogger;
+
+import com.esri.arcgisruntime.concurrent.Job;
+import com.esri.arcgisruntime.data.Feature;
+import com.esri.arcgisruntime.data.Geodatabase;
+import com.esri.arcgisruntime.data.GeodatabaseFeatureTable;
+import com.esri.arcgisruntime.data.SyncModel;
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.location.NmeaLocationDataSource;
+import com.esri.arcgisruntime.tasks.geodatabase.GenerateGeodatabaseJob;
+import com.esri.arcgisruntime.tasks.geodatabase.GenerateGeodatabaseParameters;
+import com.esri.arcgisruntime.tasks.geodatabase.GenerateLayerOption;
+import com.esri.arcgisruntime.tasks.geodatabase.GeodatabaseSyncTask;
+import com.pi4j.Pi4J;
+import com.pi4j.context.Context;
+import com.pi4j.io.serial.Baud;
+import com.pi4j.io.serial.FlowControl;
+import com.pi4j.io.serial.Parity;
+import com.pi4j.io.serial.Serial;
+import com.pi4j.io.serial.StopBits;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+public class GPS_Logger extends Application {
+
+    private Serial serial;
+    private SerialReader serialReader;
+    private final String featureLayerURL = "https://services1.arcgis.com/6677msI40mnLuuLr/arcgis/rest/services/GPS_Locations/FeatureServer";
+
+    private Geodatabase geodatabase;
+    private GeodatabaseFeatureTable table;
+    private GenerateGeodatabaseJob generateGeodatabaseJob;
+    private GeodatabaseSyncTask syncTask;
+
+    private Feature latestPosition;  // the latest position read from the GPS
+    private boolean featureUpdated = false;  // flag set to true every time we update the latestPosition from the GPS.
+
+    private Timer loggingTimer; // timer for logging to local geodatabase
+
+    public static void main(String[] args) {
+
+        Application.launch(args);
+    }
+
+    @Override
+    public void start(Stage stage) {
+
+        // set the title and size of the stage and show it
+        stage.setTitle("GPS Logger");
+        stage.setWidth(800);
+        stage.setHeight(100);
+        stage.show();
+
+        // create a JavaFX scene with a stack pane as the root node and add it to the scene
+        StackPane stackPane = new StackPane();
+        Scene scene = new Scene(stackPane);
+        stage.setScene(scene);
+
+        HBox hBox = new HBox();
+        stackPane.getChildren().add(hBox);
+
+        // Button to get offline geodatabase
+        Button btnGetGDB = new Button("Get offline gdb");
+        btnGetGDB.setOnAction(event -> downloadOfflineDB());
```

Comment:
```suggestion
        btnGetGDB.setOnAction(event -> downloadOfflineGDB());
```
Maybe renamed this method to match the others?

Reply:
I've improved the names on the button and the method